src/stylize.ts

```diff
@@ -0,0 +1,74 @@
+import {Channel} from 'vega-lite/src/channel';
+import {Type} from 'vega-lite/src/type';
+
+import {QueryConfig} from './config';
+import {SpecQueryModel} from './model';
+// import {Property} from './property';
+import {EncodingQuery, ScaleQuery} from './query/encoding';
+import {Schema} from './schema';
+import {Dict} from './util';
+
+export function stylize(answerSet: SpecQueryModel[], schema: Schema, opt: QueryConfig): SpecQueryModel[] {
+
+  answerSet = answerSet.map(function(specM) {
+    let encQIndex: Dict<EncodingQuery> = {};
+
+    if (opt.smallBandSizeForHighCardinality || opt.smallBandSizeForFacet) {
+
+      [Channel.ROW, Channel.Y, Channel.COLUMN, Channel.X].forEach((channel) => {
```

Comment:
I already mentioned that you should break sub-logic in `stylize` into multiple methods.  

Break logic in this `if` into a method.